server/src/main/java/org/eclipse/lsp/cobol/core/engine/flavors/FlavorUtils.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2021 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.core.engine.flavors;
+
+import com.google.common.collect.ImmutableMap;
+import lombok.experimental.UtilityClass;
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.tree.TerminalNode;
+import org.eclipse.lsp.cobol.core.model.SyntaxError;
+import org.eclipse.lsp.cobol.core.model.tree.Node;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/** Flavor utility class */
+@UtilityClass
+public class FlavorUtils {
+  private static final Map<String, Supplier<CobolFlavor>> FLAVOR_SUPPLIERS = ImmutableMap.of(
+      CobolFlavorImpl.NAME, CobolFlavorImpl::new
+  );
+
+  /**
+   * Process the text with flavors
+   * @param uri
+   * @param text
+   * @param flavors
+   * @return the flavor outcome
+   */
+  public FlavorOutcome process(String uri, String text, List<String> flavors) {
+    List<Node> nodes = new ArrayList<>();
+    List<SyntaxError> errors = new ArrayList<>();
+    for (String flavorName: flavors) {
+      CobolFlavor flavor = FLAVOR_SUPPLIERS.getOrDefault(flavorName, EmptyFlavor::new).get();
```

Comment:
Not a good way to handle errors from config. But it's OK for now. We don't expose flavor setting to the user, that's why we don't need to handle errors properly.